src/components/scenes/TestScene.js

```diff
@@ -0,0 +1,308 @@
+// @flow
+
+import * as React from 'react'
+import { ActivityIndicator, Alert, ScrollView, Text, View } from 'react-native'
+
+import { Gradient } from '../../modules/UI/components/Gradient/Gradient.ui'
+import { makeAaveBorrowPlugin, makeAaveDevPlugin, makeAaveKovanBorrowPlugin } from '../../plugins/borrow-plugins'
+import { type ApprovableAction, type BorrowEngine } from '../../plugins/borrow-plugins/types'
+import { filterActiveBorrowEngines, getAaveBorrowEngines } from '../../plugins/helpers/getAaveBorrowPlugins'
+import { useEffect, useState } from '../../types/reactHooks'
+import { useSelector } from '../../types/reactRedux.js'
+import { SceneWrapper } from '../common/SceneWrapper.js'
+import { TextInputModal } from '../modals/TextInputModal'
+import { Airship } from '../services/AirshipInstance'
+import { cacheStyles, getTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { MiniButton } from '../themed/MiniButton'
+
+function usePromise<T>(promise: Promise<T>): [T | void, Error | void] {
+  const [result, setResult] = useState<T | void>()
+  const [error, setError] = useState<Error | void>()
+  useEffect(() => {
+    promise.then(setResult).catch(setError)
+  }, [promise])
+  return [result, error]
+}
+
+type Props = {}
+
+const borrowPlugins = {
+  production: [makeAaveBorrowPlugin()],
+  test: [makeAaveKovanBorrowPlugin()],
+  dev: [makeAaveDevPlugin()],
+  all: [makeAaveBorrowPlugin(), makeAaveKovanBorrowPlugin(), makeAaveDevPlugin()]
+}.dev
+
+const wbtTokenId = 'd1b98b6607330172f1d991521145a22bce793277'
+const daiTokenId = 'ff795577d9ac8bd7d90ee22b6c1703490b6512fd'
+
+export function TestScene(props: Props) {
+  const theme = getTheme()
+  const styles = getStyles(theme)
+  const account = useSelector(state => state.core.account)
+  const [whenBorrowEngines, setWhenBorrowEngines] = useState(getAaveBorrowEngines(borrowPlugins, account).then(filterActiveBorrowEngines))
```

Comment:
Ah yes. Good catch.

Reply:
Not sure if the `useState(() => ...)` is valid syntax. Maybe `useMemo` or `useCallback` is the correct hook for this job. Either way, it doesn't matter because this is just a test scene.